x/paychan/codec.proto

```diff
@@ -0,0 +1,76 @@
+syntax = "proto3";
+
+package paychan;
+
+import "github.com/iov-one/weave/x/codec.proto";
+import "github.com/iov-one/weave/crypto/models.proto";
+
+// PaymentChannel holds the state of a payment channel during its lifetime.
+message PaymentChannel {
+    // Sender is the source that the founds are allocated from.
+    crypto.PublicKey sender_public_key = 1;
+    bytes recipient = 2;
+    // Total represents a maximum value that can be transferred via this
+    // payment channel.
+    x.Coin total = 3;
```

Comment:
I support only one because the implementation is simpler. If it makes sense, I can change it into a collection of coins. 
Escrow is using multiple coins.

Reply:
As discussed on slack, I will leave it single currency to keep the logic simpler.